app/views/map/map.html.erb

```diff
@@ -28,9 +28,21 @@
         new L.LatLng(-54.36775852, 178.59375)
       );
 
-      var lat = parseFloat(params.lat)|| 23;
-      var lon = parseFloat(params.lon) || 77;
-      var zoom = parseFloat(params.zoom) || 4;
+      var user_lat = null;
```

Comment:
Hey, i think we should do all these calculations in the controller. From the controller we should pass only lat, lon, zoom, and layer_name. Thanks!

Reply:
Ahhh, that's a good point. Can the controller access the urlHash though? It's in Javascript. I'll take a look at it all and see what I can move to the controller!

Reply:
I think we can't... Without reimplementing urlhash in Ruby w request.fullpath maybe? Do you think it's worth it? What are the pros and cons?

Reply:
My thought is that the Controller is only called once on page load, and the url hash is accessed many times in the map. It makes sense to keep the url hash calculations all on one page.

I do see some other things that can be cleaned up and moved to the controller though!